file_catalog/server.py

```diff
@@ -627,20 +583,15 @@ def initialize(self, **kwargs: Any) -> None:  # type: ignore[override]  # pylint
         # pylint: disable=W0201
         self.files_url = os.path.join(self.base_url, 'files')
 
-    @handler.authenticated  # type: ignore[misc]
-    @handler.catch_error  # type: ignore[misc]
+    @fc_auth(prefix=FC_AUTH_PREFIX, roles=FC_AUTH_ROLES)
     async def post(self, uuid: str) -> None:
         """Handle POST request.
 
         Remove location from the record identified by the provided UUID,
         and potentially the entire record.
         """
         # try to load the record from the file catalog by UUID
-        try:
-            db_file = await self.db.get_file({'uuid': uuid})
-        except pymongo.errors.InvalidId:
```

Comment:
Is this no longer a concern?

Reply:
I think it will never raise that error, since `InvalidId` is only for the `_id` field.

Reply:
Yeah, `pymongo.errors.InvalidId` no longer exists. (I think I had to go back to pymongo v1 to find it.)
So it doesn't really make sense to catch it any more.

There is a BSON version of the error, but I think it's like David says, only when Mongo has some internal trouble with `_id`.

That `db` is the file catalog's internal MongoDB client, so if we really want to do something about Mongo errors, it's probably better handled at that level. 

The keycloak auth decorator includes `catch_error`. For now, letting an exception bubble up and cause an error should work fine.